query/api/src/main/java/org/codice/ddf/admin/api/report/Report.java

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ **/
+package org.codice.ddf.admin.api.report;
+
+import java.util.List;
+
+public interface Report {
```

Comment:
We can't get rid of `FunctionReport` because it's part of the `FunctionField` interface that requires a `DataType` as a value, any other type of value will likely break graphql. We could get rid of `ReportWithResult` and `Report` but the impls will have identical methods to the `FunctionField` interface. I think we should keep it the way it for now, those report interfaces could be handle for communicating across services.